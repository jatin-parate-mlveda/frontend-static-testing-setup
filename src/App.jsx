import { Route, Routes, NavLink } from 'react-router-dom';
import { ErrorBoundary } from 'components/ErrorBoundary';
import classes from './App.module.scss';
import SuspendedBlogDetail from './pages/blog-detail/SuspendedBlogDetail';
import SuspendedHomePage from './pages/home/SuspendedHomePage';

export default function App() {
  return (
    <ErrorBoundary>
      <div className={classes.app}>
        <header className={classes.app__header}>
          <NavLink className={classes.app__link} to='/'>
            Home
          </NavLink>
          <NavLink className={classes.app__link} to='/blog-detail'>
            Blog
          </NavLink>
        </header>
        <Routes>
          <Route path='/blog-detail' element={<SuspendedBlogDetail />} />
          <Route index element={<SuspendedHomePage />} />
          <Route path='*' element={<h1>Page not found!</h1>} />
        </Routes>
      </div>
    </ErrorBoundary>
  );
}
