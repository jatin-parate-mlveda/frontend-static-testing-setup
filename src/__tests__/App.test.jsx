import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import classes from '../App.module.scss';
import { waitFor2Sec } from '../../test/utils/wait-for-2sec';

test('It renders loading state successfully', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );
  const eles = container.getElementsByClassName(classes.app);

  expect(eles).toHaveLength(1);

  expect(container.outerHTML).toMatchSnapshot();
});

test('It renders successfully', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );

  await waitFor2Sec();

  const eles = container.getElementsByClassName(classes.app);

  expect(eles).toHaveLength(1);

  expect(container.outerHTML).toMatchSnapshot();
});

test('It renders blogs page loading state', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={[{ pathname: '/blog-details' }]}>
      <App />
    </MemoryRouter>,
  );

  expect(container.outerHTML).toMatchSnapshot();
});

test('It renders blogs page successfully', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={[{ pathname: '/blog-details' }]}>
      <App />
    </MemoryRouter>,
  );

  await waitFor2Sec();

  const eles = container.getElementsByClassName(classes.app);

  expect(eles).toHaveLength(1);

  expect(container.outerHTML).toMatchSnapshot();
});
