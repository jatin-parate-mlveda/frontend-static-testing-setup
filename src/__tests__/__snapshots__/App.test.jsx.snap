// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders blogs page loading state 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a class=\\"app__link\\" href=\\"/\\">Home</a><a class=\\"app__link\\" href=\\"/blog-detail\\">Blog</a></header><h1>Page not found!</h1></div></div>"`;

exports[`It renders blogs page successfully 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a class=\\"app__link\\" href=\\"/\\">Home</a><a class=\\"app__link\\" href=\\"/blog-detail\\">Blog</a></header><h1>Page not found!</h1></div></div>"`;

exports[`It renders loading state successfully 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a aria-current=\\"page\\" class=\\"app__link active\\" href=\\"/\\">Home</a><a class=\\"app__link\\" href=\\"/blog-detail\\">Blog</a></header><h1>Loading...</h1></div></div>"`;

exports[`It renders successfully 1`] = `"<div><div class=\\"app\\"><header class=\\"app__header\\"><a aria-current=\\"page\\" class=\\"app__link active\\" href=\\"/\\">Home</a><a class=\\"app__link\\" href=\\"/blog-detail\\">Blog</a></header><div class=\\"container\\"><h1>Hello world!</h1></div></div></div>"`;
